src/index.ts

```diff
@@ -1,131 +1,155 @@
-    check: (action: string, resource: string)=>boolean;
-    addKeyToState: (action: string, resource: string)=>Promise<void>;
-    loadLocalState: (actionsResourcesList: ActionResourceSchema[])=>Promise<void>;
-    getCaslJson: ()=>CaslPermissionSchema[];
-    loadLocalStateBulk: (actionsResourcesList: ActionResourceSchema[])=>Promise<void>;
+  loggedInUser: string;
+  backendUrl: string;
+  defaultAnswerIfNotExist: boolean;
+  state: PermitStateSchema;
+  check: (action: string, resource: string, attributes?: Record<string, any>) => boolean; // Added attributes
+  addKeyToState: (action: string, resource: string, attributes?: Record<string, any>) => Promise<void>; // Added attributes
+  loadLocalState: (actionsResourcesList: ActionResourceSchema[]) => Promise<void>;
+  getCaslJson: () => CaslPermissionSchema[];
+  loadLocalStateBulk: (actionsResourcesList: ActionResourceSchema[]) => Promise<void>;
 }
 
 export interface CaslPermissionSchema {
-    action: string;
-    subject: string;
-    inverted: boolean; // if true, the permission is denied
+  action: string;
+  subject: string;
+  inverted: boolean; // if true, the permission is denied
 }
 
 export interface PermitStateSchema {
-    [key: string]: boolean;
+  [key: string]: boolean;
 }
 
 export interface ActionResourceSchema {
-    action: string;
-    resource: string;
+  action: string;
+  resource: string;
+
+  attributes?: Record<string, any>;
 }
 
 const getBulkPermissionFromBE = async (url: string, user: string, actionsResourcesList: ActionResourceSchema[]): Promise<boolean[]> => {
-    return await axios.post(`${url}?user=${user}`,{resourcesAndActions:actionsResourcesList}).then(response => {
-        return response.data.permittedList;
-   });
-}
-const getPermissionFromBE = async (url: string, user: string, action: string, resource: string, defaultPermission: boolean): Promise<boolean> => {
-    return await axios.get(`${url}?user=${user}&action=${action}&resource=${resource}`).then(response => {
-        return response.data.permitted;
-    }
-    ).catch(error => {
-        if (error.response.status === 403) {
-            return false;
-        }
-        // tslint:disable-next-line:no-console
-        console.error(error);
-        return defaultPermission;
+  const payload = actionsResourcesList.map((actionResource) => {
+    return {
+      action: actionResource.action,
+      resource: actionResource.resource,
+      attributes: actionResource.attributes || {},
+    };
+  });
+  return await axios.post(`${url}?user=${user}`, { resourcesAndActions: payload }).then((response) => {
+    return response.data.permittedList;
+  });
+};
+const getPermissionFromBE = async (
+  url: string,
+  user: string,
+  action: string,
+  resource: string,
+  defaultPermission: boolean,
+  attributes: Record<string, any> = {},
+): Promise<boolean> => {
+  const attributeQuery = attributes && Object.keys(attributes).length > 0 ? `&attributes=${JSON.stringify(attributes)}` : '';
```

Comment:
You can have a lot of attributes and it is better to pass it with a POST request. Still, we also don't want to break the old version that uses GET, so I think the best way is to create getPermissionWithAttrs with POST instead of GET and loadLocalStateWithAttrs that will use it, what do you think?

Reply:
I'd do that and remove the logic from GET

Reply:
but this loadLocalStateWithAttrs is not required because the loadLocalStateBulk inside is backwards compatible. I have removed them from GET logic

Reply:
So you suggest that attrs will be supported only for bulk actions?

Reply:
I think if someone wants to just evaluate only single permissions, they just send only single object in the array. I think created multiple endpoints is redundant here